src/data-model-providers/codegen/CodegenDataModelProvider_Read.cpp

```diff
@@ -116,27 +114,13 @@ DataModel::ActionReturnStatus CodegenDataModelProvider::ReadAttribute(const Data
     }
 
     // Read via AAI
-    std::optional<CHIP_ERROR> aai_result;
-    if (const EmberAfCluster ** cluster = std::get_if<const EmberAfCluster *>(&metadata))
-    {
-        Compatibility::GlobalAttributeReader aai(*cluster);
-        aai_result = TryReadViaAccessInterface(request.path, &aai, encoder);
-    }
-    else
-    {
-        aai_result = TryReadViaAccessInterface(
-            request.path, AttributeAccessInterfaceRegistry::Instance().Get(request.path.mEndpointId, request.path.mClusterId),
-            encoder);
-    }
+    std::optional<CHIP_ERROR> aai_result = TryReadViaAccessInterface(
+        request.path, AttributeAccessInterfaceRegistry::Instance().Get(request.path.mEndpointId, request.path.mClusterId), encoder);
     VerifyOrReturnError(!aai_result.has_value(), *aai_result);
 
-    if (!std::holds_alternative<const EmberAfAttributeMetadata *>(metadata))
-    {
-        // if we only got a cluster, this was for a global attribute. We cannot read ember attributes
-        // at this point, so give up (although GlobalAttributeReader should have returned something here).
-        chipDie();
-    }
     const EmberAfAttributeMetadata * attributeMetadata = std::get<const EmberAfAttributeMetadata *>(metadata);
+    // We can only get a status or metadata.
```

Comment:
What guarantees that?  What if someone in fact does a read for a global attribute on the provider?

Reply:
I have changed FindAttributeMetadata to not return the cluster path anymore as global attribute handling is done somewhere else.

What you would get if you actually try it would be `UnsupportedAttribute` as we only process attributes in metadata.

I will update documentation for the provider interface to say explicitly that the "list" attributes are not to be handled.